internal/edgedbtypes/datetime.go

```diff
@@ -0,0 +1,47 @@
+// This source file is part of the EdgeDB open source project.
+//
+// Copyright 2020-present EdgeDB Inc. and the EdgeDB authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package edgedbtypes
+
+import (
+	"time"
+)
+
+const timeShift = 946_684_800
+
+// NewLocalDateTime returns a new LocalDateTime
+func NewLocalDateTime(
+	year int, month time.Month, day, hour, minute, second, microseconds int,
+) LocalDateTime {
+	t := time.Date(
+		year, month, day, hour, minute, second, microseconds*1_000, time.UTC,
+	)
+	sec := t.Unix() - timeShift
+	nsec := int64(t.Sub(time.Unix(t.Unix(), 0)))
+	return LocalDateTime{sec*1_000_000 + nsec/1_000}
+}
+
+// LocalDateTime is a date and time without timezone.
+// https://www.edgedb.com/docs/datamodel/scalars/datetime/
+type LocalDateTime struct {
+	usec int64
```

Comment:
@1st1 @tailhook @elprans I updated this with the feedback from https://github.com/edgedb/edgedb-go/pull/90#discussion_r569736492 let me know if I am on the right track.

Reply:
So this is now a proper unix timestamp?

Reply:
No it is still a 2000's time stamp but that is an implementation detail except for the zero value. :thinking: i guess `LocalDateTime{} == "1970-01-01T00:00"` makes more sense than `"2000-01-01T00:00"`. I can change that. What about the rest of the public API?

Reply:
> No it is still a 2000's time stamp but that is an implementation detail except for the zero value. 

The zero value is what bothers me. I'd expect it to be `00/00/0000`, not some weird 2000s date.

Also the argument about `9999/9999/9999` kind of values is IMO invalid. The max possible sane LocalDate would be `9999/12/31`.

Reply:
IOW I'm not so sure we should use the timestamp internally. This type, again, is more about wall calendar, than about all possible future/past timelines. @elprans @tailhook

Reply:
I have no problem with using a timestamp internally as long as it is opaque, i.e. there should not be a (documented?) way of setting or getting the raw value from user code.  The timestamp approach has advantages: it's memory-efficient and makes comparisons easy and fast.

Reply:
> I have no problem with using a timestamp internally as long as it is opaque, i.e. there should not be a (documented?) way of setting or getting the raw value from user code.

Well, that's the crux of the problem. It will be opaque, unless you just create an empty struct, like `edgedb.LocalDate{}`, in which case the local date you'd get wouldn't be `0/0/0`, it would be `2000/01/30` or something.

Reply:
> It will be opaque, unless you just create an empty struct, like edgedb.LocalDate{}, in which case the local date you'd get wouldn't be 0/0/0, it would be 2000/01/30 or something.

We can make the zero value be `0/0/0` no problem.

Reply:
Let's keep it simple as Elvis & Paul suggested, with one field in the struct.